src/app/item-list/item-list.component.ts

```diff
@@ -58,50 +74,36 @@ export class ItemListComponent implements OnInit {
       });
   }
 
-  getBook(id: number): void{
-    this.bookService.getBook(id)
-        .then(book => this.selectedBook);
-        
-  }
-
   ngOnInit(): void {
-    this.getItems(this.url);
+    // console.log('ngOnInit');
+    this.getItems();
   }
 
-  // onSelect(book: Book): void{
-  //   this.selectedBook = book;
-  // }
-
   capitalizeFirstLetter(string) {
     return string.charAt(0).toUpperCase() + string.slice(1);
   }
 
   goToDetail(id: number): void{
     const detailCollectionUrl = this.url[1].slice(0,-1);
-    const detailChildCollectionUrl = this.url[1];
-    if(this.url[1] === 'books'){
-      this.router.navigate([`${detailCollectionUrl}`, id]);
-    }
-    else if(this.url[1] === 'stores'){
-      this.router.navigate([`${detailCollectionUrl}`, id]);
-    }
-    else if(this.url[1] === 'store'){
-      const storeId = this.url[2];
-      //console.log(storeId);
-      this.router.navigate([`${detailChildCollectionUrl}/${+storeId}/book`, id]);
+    let navigateToUrl = '';
+    switch(this.listCollectionUrl){
+      case 'books':
+        navigateToUrl = `${detailCollectionUrl}`;
```

Comment:
`books` and `stores` can be covered by a single `case` statement using an expression. Look at  https://stackoverflow.com/questions/3463833/expression-inside-switch-case-statement
 for ideas